discovery-provider/src/queries/notifications.py

```diff
@@ -201,6 +198,90 @@ def get_cosign_remix_notifications(session, max_block_number, remix_tracks):
     return remix_notifications
 
 
+class GroupMilestones(TypedDict):
+    tracks: Dict[int, int]
+    albums: Dict[int, int]
+    playlists: Dict[int, int]
+
+
+class MilestoneInfo(TypedDict):
+    favorite_counts: GroupMilestones
+    repost_counts: GroupMilestones
+    follower_counts: Dict[int, int]
+
+
+def get_milestone_info(
+    session: Session, min_block_number, max_block_number
+) -> MilestoneInfo:
+    milestones_in_block = session.query(
+        Milestone.name, Milestone.id, Milestone.threshold
+    ).filter(
+        Milestone.blocknumber > min_block_number,
+        Milestone.blocknumber <= max_block_number,
+    )
+    milestones = defaultdict(list)
+
+    for milestone_name, *id_threshold in milestones_in_block:
+        milestones[milestone_name].append(id_threshold)
+
+    milestone_playlist_ids = [
+        playlist_id
+        for name, playlist_id, _ in milestones_in_block
+        if name
+        in (
+            MilestoneName.PLAYLIST_REPOST_COUNT,
+            MilestoneName.PLAYLIST_SAVE_COUNT,
+        )
+    ]
+
+    playlist_id_albums: Dict[int, bool] = {}
+    if milestone_playlist_ids:
+        playlist_id_albums_response = (
+            session.query(Playlist.playlist_id, Playlist.is_album)
+            .filter(
+                Playlist.is_current == True,
+                Playlist.playlist_id.in_(milestone_playlist_ids),
+            )
+            .all()
+        )
+        playlist_id_albums = dict(playlist_id_albums_response)
+
+    album_favorites: List[Tuple[int, int]] = []
+    playlist_favorites: List[Tuple[int, int]] = []
+    for id, threshold in milestones.get(MilestoneName.PLAYLIST_SAVE_COUNT, []):
```

Comment:
I guess trigger could set `ALBUM_SAVE_COUNT`, but I think this is fine for now because the future notification table can capture this info, right?

Reply:
Yes, I don't think we should differentiate between album and playlist here, and yes eventually it'll be in the notifications tbl too